kanidm_book/src/security_hardening.md

```diff
@@ -89,25 +93,29 @@ is to set configurations to:
 The kanidm server should be run as "kanidm:kanidm" with the appropriate user and user private
 group created on your system. This applies to unixd configuration as well.
 
+TODO is the kanidm user created the same way as other daemon users, without a /home dir and 
```

Comment:
Correct (typically with systemd which dynamically creates the user and its associated uid, if I understand it properly) 👍🏻

Reply:
Yep, that's right, systemd dynamically creates it

Reply:
I need more clarification, please. How does this work, does systemd create the user & UID at installation, and then it is always the same? (I am not testing anything yet, just reviewing for clarity & typos)

Reply:
Ahhh it's much more magical, just like anything with systemd involved. 

http://0pointer.net/blog/dynamic-users-with-systemd.html

What happens is that when systemd starts up the service, it dynamically generates a uid/gid, and creates a home directory for the user. When the service stops the user "vanishes". This way, you can have the kanidm account appear/disappear basicly, because systemd will auto-magically chown your home dir and all that.


In the docker case, it's up to the admin to create and nominate a uid/gid to run the server as to provide to docker or their container engine.

Reply:
Thanks! My #1 question is usually "Do I have to do something, or is there magick and I do not have to do something"